iroha-lib/model/generators/QueryGenerator.cpp

```diff
@@ -0,0 +1,161 @@
+    queryPayload->set_account_id(account_id);
+    return query;
+}
+
+std::shared_ptr<iroha::protocol::Query> QueryGenerator::generateGetAccountDetail(const std::string& account_id,
+                                                                                 uint64_t counter,
+                                                                                 const uint64_t created_time)
+{
+    auto query = generateQuery(account_id, counter, created_time);
+    auto queryPayload = query->mutable_payload()->mutable_get_account_detail();
+    queryPayload->set_account_id(account_id);
+    return query;
+}
+
+std::shared_ptr<iroha::protocol::Query> QueryGenerator::generateGetAccountTransactions(const std::string& account_id,
+                                                                                       uint64_t counter,
+                                                                                       const uint64_t created_time,
+                                                                                       const std::optional<std::string*>& first_tx_hash,
+                                                                                       const std::optional<google::protobuf::Timestamp*>& first_tx_time,
+                                                                                       const std::optional<google::protobuf::Timestamp*>& last_tx_time,
+                                                                                       const std::optional<uint64_t> first_tx_height,
+                                                                                       const std::optional<uint64_t> last_tx_height)
+{
+    auto query = generateQuery(account_id, counter, created_time);
+    auto queryPayload = query->mutable_payload()->mutable_get_account_transactions();
+    queryPayload->set_account_id(account_id);
+    if (first_tx_hash.has_value())
+        query->mutable_payload()->mutable_get_account_transactions()->mutable_pagination_meta()->set_allocated_first_tx_hash(first_tx_hash.value());
+    if (first_tx_time.has_value())
+        query->mutable_payload()->mutable_get_account_transactions()->mutable_pagination_meta()->set_allocated_first_tx_time(first_tx_time.value());
+    if (last_tx_time.has_value())
+        query->mutable_payload()->mutable_get_account_transactions()->mutable_pagination_meta()->set_allocated_last_tx_time(last_tx_time.value());
+    if (first_tx_height.has_value())
+        query->mutable_payload()->mutable_get_account_transactions()->mutable_pagination_meta()->set_first_tx_height(first_tx_height.value());
+    if (last_tx_height.has_value())
+        query->mutable_payload()->mutable_get_account_transactions()->mutable_pagination_meta()->set_last_tx_height(last_tx_height.value());
+    return query;
+}
+
+std::shared_ptr<iroha::protocol::Query> QueryGenerator::generateGetAccountAssetTransactions(const std::string& account_id,
```

Comment:
More reasons to use a different style of bracing.

Reply:
Done.